cmd/erasure-server-pool.go

```diff
@@ -272,10 +283,37 @@ func (z *erasureServerPools) getPoolIdxExisting(ctx context.Context, bucket, obj
 			// exist proceed to next pool.
 			continue
 		}
-		return i, nil
+
+		// When versionID is specified we look for additional errors
+		if opts.VersionID != "" {
+			if isErrMethodNotAllowed(errs[i]) || isErrVersionNotFound(errs[i]) {
+				// No object exists or its a delete marker,
+				// check objInfo to confirm.
+				if objInfos[i].DeleteMarker && objInfos[i].Name != "" {
```

Comment:
Can objInfos[i] be valid when errs[i] is not nil ?

Reply:
yes @vadmeste for delete markers